### PR TITLE
Add logical_and function to paddle.tensor.logic

### LINKED ISSUE #15784: logical_and

### PR DESCRIPTION
**Pull request by KSANTHOSH200**
Close #15784

### CODE REVIEW

File: ivy/functional/frontends/paddle/tensor/logic.py [hunk 1]
```diff
@@ -75,3 +75,11 @@ def less_than(x, y, /, *, name=None):
 @to_ivy_arrays_and_back
 def less_equal(x, y, /, *, name=None):
     return ivy.less_equal(x, y)
+
+@with_unsupported_dtypes(
+    {"2.4.2 and below": ("bool", "uint8", "int8", "int16", "complex64", "complex128")},
```
Comment (rishabgit) [inline ivy/functional/frontends/paddle/tensor/logic.py:81]: Please update the supported dtypes according to the paddle's doc - https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/logical_and_en.html

File: ivy/functional/frontends/paddle/tensor/stat.py [hunk 2]
```diff
@@ -9,3 +9,5 @@
 @to_ivy_arrays_and_back
 def mean(input, axis=None, keepdims=False, out=None):
     return ivy.std(input, axis=axis, keepdims=keepdims, out=None)
+
```
Comment (rishabgit) [inline ivy/functional/frontends/paddle/tensor/stat.py]: Extra new line here isn't needed. pre-commit should help auto fix these style issues - https://unify.ai/docs/ivy/overview/contributing/setting_up.html#pre-commit

**Comment by rishabgit** [comment]
ivy-gardener

**Review by rishabgit — CHANGES_REQUESTED** [review]
Hi! Thanks for the PR :) Requested a couple of changes

File: ivy/functional/frontends/paddle/tensor/logic.py (hunk 1 above)
Comment (KSANTHOSH200) [reply ivy/functional/frontends/paddle/tensor/logic.py]: Made the changes. Please review.

**Comment by rishabgit** [comment]
Hey @KSANTHOSH200, thanks for making those changes. It looks like the tests are failing, could you look into it?
 You can find the errors in Determine and Run tests tab of [intelligent-tests-pr ](https://github.com/unifyai/ivy/actions/runs/5056657263/jobs/9074394003?pr=15785) 

Also, here's a quick way to run the tests on local before pushing the changes - 
```
pytest ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_logic.py::test_paddle_logical_and
```

**Comment by rishabgit** [comment]
> Hi @rishabgit,
> 
> I think the test is failing because of library than of code.
> 
> Thanks

Hi! I quickly tested the code, and it seems to work fine - 
```
import ivy.functional.frontends.paddle as paddle_frontend
import paddle
import ivy

x = ivy.Array(paddle.to_tensor([True]))
y = ivy.Array(paddle.to_tensor([True, False, True, False]))

paddle_frontend.logical_and(x, y)
# ivy.frontends.paddle.Tensor([ True, False,  True, False])
```

Could you look into whether the test is set up correctly? Hope this helps - https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends_tests.html

**Pull request closed without merging**